### ISSUE TITLE
LDAP Config Test Passes But Login Page Fails

### ISSUE THREAD

**Issue by ridius** [instructure/canvas-lms#909]
### Summary:
I updated to the most recent stable build today (commit 692f98b875dc672623dabb6d5344b1f5fe4f6118) and tried to update my LDAP Bind account password (I recently had to change it due to IT policy). I successfully configured the LDAP account in Authentication Settings, successfully tested LDAP integration, and set LDAP as the 1st provider. I thought everything was great, but none of my users can login via LDAP. The login page always says Invalid Username or Password and I get an HTTP 400 return after trying to login. I don't see any error messages in the Canvas logs or the Nginx logs. I can login with the Canvas account.

My environment:

- Ubuntu 14.04.5 LTS
- Ruby 2.2.5p319
- NodeJS 6.7.0
- Nginx 1.10.1
- Phusion Passenger 5.0.30 

### Steps to reproduce:

 1. Configure LDAP integration settings
 2. Successfully test LDAP
 3. Attempt to login from the login page

### Expected behavior:
LDAP credentials successfully login.

### Additional notes:

**Comment by ccutrer** [comment]
check the following:
 * there is an already existing user and login in Canvas that matches the username you're using with LDAP (or the LDAP attribute specified by the "Login Attribute" in the Canvas LDAP Config)
 * said login in Canvas isn't explicitly associated with the Canvas authentication provider (this is not possible or visible via the UI, so would only happen if you created the login with an SIS Import or via API, and manually specified 'canvas' for the provider)

**Comment by ridius** [comment]
Thanks @ccutrer,

These users definitely already have user accounts in Canvas. They were all working before the upgrade I did today. Most of these user accounts were created through an SIS Import, but I never specified a specific authentication provider as a part of that import.

I tried doing an SIS Export of the Users, but the resulting CSV did not specify the authentication provider attribute of the user profiles.
